src/main/java/mate/zorii/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -33,4 +34,19 @@ public BookDto findById(Long id) {
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(
                 () -> new EntityNotFoundException("No book found by id " + id)));
     }
+
+    @Override
+    public BookDto update(Long id, CreateOrUpdateBookRequestDto requestDto) {
+        if (bookRepository.existsById(id)) {
```

Comment:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("No book with id " + id + " found.")));
        bookMapper.updateBookFromDto(bookDto, book);
        return bookMapper.toDto(bookRepository.save(book));
```

Create updateBookFromDto method in BookMapper